pkg/controllers/core/deployer_controller.go

```diff
@@ -396,11 +473,55 @@ func (r *DeployerReconciler) reconcileChildService(ctx context.Context, log logr
 	return service, nil
 }
 
+func (r *DeployerReconciler) ingressSemanticEquals(desiredIngress, ingress *v1beta1.Ingress) bool {
+	return equality.Semantic.DeepEqual(desiredIngress.Spec, ingress.Spec) &&
+		equality.Semantic.DeepEqual(desiredIngress.ObjectMeta.Labels, ingress.ObjectMeta.Labels)
+}
+
 func (r *DeployerReconciler) serviceSemanticEquals(desiredService, service *corev1.Service) bool {
 	return equality.Semantic.DeepEqual(desiredService.Spec, service.Spec) &&
 		equality.Semantic.DeepEqual(desiredService.ObjectMeta.Labels, service.ObjectMeta.Labels)
 }
 
+func (r *DeployerReconciler) constructIngressForDeployer(deployer *corev1alpha1.Deployer) (*v1beta1.Ingress, error) {
+	labels := r.constructLabelsForDeployer(deployer)
+
+	ingress := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:       labels,
+			Annotations:  make(map[string]string),
+			GenerateName: fmt.Sprintf("%s-deployer-", deployer.Name),
+			Namespace:    deployer.Namespace,
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: fmt.Sprintf("%s.%s", deployer.Name, domain),
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
```

Comment:
for ingress controllers that support https, do we need to define a rule, or will they handle that automatically?

It's also fine to punt on this.